app/models/spree/subscription.rb

```diff
@@ -226,10 +243,13 @@ def add_shipping_costs_to_order(order)
       end
 
       def add_payment_method_to_order(order)
+        recurring_payment_method = Spree::Gateway::StripeSubscription.find_by(name: 'Credit Card', active: true)
+        raise "missing stripe recurring payment method" unless recurring_payment_method
+        
         if order.payments.exists?
-          order.payments.first.update(source: source, payment_method: source.payment_method)
+          order.payments.first.update(source: source, payment_method: recurring_payment_method)
```

Comment:
what if it fails?

order.payments.first, will it not point to first time payment done when order first created? will it not update that payment rather than creating a new one?

Reply:
First of all this is the gem code, order doesnt refer to parent order, parent order exist with the variable named parent order. This order represents new order which we are generating